src/contents/ui/main.qml

```diff
@@ -48,12 +66,14 @@ PlasmoidItem {
     compactRepresentation: Item {
         id: compact
 
-        Layout.preferredWidth: grid.implicitWidth + Kirigami.Units.smallSpacing * 2
-        Layout.preferredHeight: grid.implicitHeight + Kirigami.Units.smallSpacing * 2
+        Layout.preferredWidth: grid.implicitWidth + (horizontal ? lengthMargin : 0)
+        Layout.preferredHeight: grid.implicitHeight + (horizontal ? 0 : lengthMargin)
         Layout.fillHeight: horizontal
         Layout.fillWidth: !horizontal
 
-        readonly property real controlsSize: Math.min(height, Kirigami.Units.iconSizes.medium)
+        readonly property int widgetThickness: Math.min(height, width)
+        readonly property int controlsSize: Math.round(widgetThickness * 0.75)
+        readonly property int lengthMargin: Math.round((widgetThickness - controlsSize))
```

Comment:
I would prefer something like this, so that the controlSize is limited to `Kirigami.Units.iconSizes.medium` and the margin limited to `Kirigami.Units.smallSpacing * 2`.
```suggestion
        readonly property int widgetThickness: Math.min(height, width)
        readonly property int lengthMargin: Math.min(Math.round(widgetThickness * 0.2), Kirigami.Units.smallSpacing * 2)
        readonly property int availableWidgetThickness: widgetThickness - lengthMargin
        readonly property int controlsSize: Math.min(availableWidgetThickness, Kirigami.Units.iconSizes.medium)
```

Reply:
Done, but

Personally, I prefer the image and icons to update with the panel at all sizes, it looks more consistent with other widgets that normally grow as big as the panel. Also same margin on all edges looks better for the tinted background IMHO.

Maybe we can have this as an option?

Reply:
Mmm, I made some tests with different panel sizes and indeed you're right, your first implementation was better :sweat_smile:

Can you restore the first version? :pray:

Reply:
A further improvement would be to make the margin/control-size ratio configurable (the 0.75 that is now hardcoded). But it can also be implemented in an other PR.

Reply:
> Can you restore the first version?

Done